pkg/commands/auth/run.go

```diff
@@ -0,0 +1,77 @@
+package auth
+
+import (
+	"context"
+	"os"
+
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/types"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/pkg/flag"
+	"github.com/aquasecurity/trivy/pkg/log"
+)
+
+func Login(_ context.Context, registry string, opts flag.Options) error {
+	if len(opts.Credentials) == 0 {
+		return xerrors.New("username and password required")
+	} else if len(opts.Credentials) > 1 {
+		return xerrors.New("multiple credentials are not allowed")
+	}
+
+	reg, err := name.NewRegistry(registry)
+	if err != nil {
+		return xerrors.Errorf("failed to parse registry: %w", err)
+	}
+	serverAddress := reg.Name()
+
+	cf, err := config.Load(os.Getenv("DOCKER_CONFIG"))
+	if err != nil {
+		return xerrors.Errorf("failed to load docker config: %w", err)
+	}
+	creds := cf.GetCredentialsStore(serverAddress)
+	if serverAddress == name.DefaultRegistry {
+		serverAddress = authn.DefaultAuthKey
+	}
+	if err := creds.Store(types.AuthConfig{
+		ServerAddress: serverAddress,
+		Username:      opts.Credentials[0].Username,
+		Password:      opts.Credentials[0].Password,
+	}); err != nil {
+		return xerrors.Errorf("failed to store credentials: %w", err)
+	}
+
+	if err := cf.Save(); err != nil {
```

Comment:
`docker login` checks credentials:
```bash
➜ cat pass.txt | docker login --username foo --password-stdin ghcr.io

Error response from daemon: Get "https://ghcr.io/v2/": denied: denied
```
We don't do that:
```bash
➜ cat pass.txt | ./trivy auth login --username foo --password-stdin ghcr.io

2024-10-08T12:50:55+06:00	INFO	Logged in	file_path="config/config.json"
```
Users will expect the same behavior.
We need to add this check (if possible) or write in the docs that Trivy just records the credentials without checking.

Reply:
I was aware of the difference, but `crane auth login` doesn't validate the credential, and I complied with it. However, it's better to do that in terms of UX, as you said. Added https://github.com/aquasecurity/trivy/pull/7664/commits/b5be50fa81ae9d88573270a75cd17b8caf366944.